network_diffusion/models/ICM_model2.py

```diff
@@ -0,0 +1,173 @@
+
+from typing import Dict, List, Tuple
+import random
+import numpy as np
+import networkx as nx
+from network_diffusion.mln.mlnetwork import MultilayerNetwork
+from network_diffusion.models.base_model import BaseModel
+from network_diffusion.mln.actor import MLNetworkActor
+from network_diffusion.models.utils.compartmental import CompartmentalGraph
+from network_diffusion.models.utils.types import NetworkUpdateBuffer as NUBff
+from network_diffusion.seeding.base_selector import BaseSeedSelector
+from network_diffusion.utils import BOLD_UNDERLINE, THIN_UNDERLINE, NumericType
+
+
+class ICModel(BaseModel):
+
+    INACTIVE_NODE = "0"
+    ACTIVE_NODE = "1"
+    PROCESS_NAME = "ICM"
+    DONE = "-1"
```

Comment:
I'd change states to:
- inactive
- active
- activated

With transitions 
inactive->active with probability p (passed in the __init__)
active->activated with probability 1 (this is an equivalent of your "done")